spark/src/main/scala/io/ddf/spark/content/ViewHandler.scala

```diff
@@ -67,29 +67,32 @@ class ViewHandler(mDDF: DDF) extends io.ddf.content.ViewHandler(mDDF) with IHand
       throw new IllegalArgumentException("Number of samples must be larger than or equal to 0")
     }
 
-    val sampleDDF = if (!withReplacement) {
-      if (seed >= 0) {
-        mDDF.getSqlHandler.sql2ddf(s"select * from ${mDDF.getSchema.getTableName} order by rand($seed) limit $numSamples")
-      } else {
-        // This is a workaround to deal with Spark's inability to order rows by rand(<negative_seed>)
-        // Example: select * from table order by rand(-123)
-        mDDF.getSqlHandler.sql2ddf(s"select ${mDDF.getColumnNames.mkString(",")} from " +
-          s"(select *, rand($seed) as rnd from ${mDDF.getSchema.getTableName}) t order by rnd limit $numSamples")
-      }
-    } else {
+    val numRows = mDDF.getNumRows
+    var fraction = 1.0
+    // We use Spark's API to sample twice what we need, then only pick the first numSamples rows.
+    if (withReplacement) fraction = 2.0 * numSamples / numRows
+
+    val manager = this.getManager
+    val schema: Schema = new Schema(null,
+      JavaConverters.asScalaBufferConverter(mDDF.getSchema.getColumns).asScala.toArray)
+    // Optimized running time for integer values by using Spark DataFrame sample and limit APIs
+    val numSamplesInt = numSamples.toInt
+    val sampleDDF = if (numSamplesInt == numSamples) {
```

Comment:
Good point. I checked and saw that the casted value is always of opposite sign. 
For example
`scala> -111111111111L.toInt
res51: Int = 558038585

scala> -111111L.toInt
res52: Int = -111111

scala> 111111111111L.toInt
res53: Int = -558038585

scala> 1111111L.toInt
res54: Int = 1111111`
BTW, do you think there is a cast in `numSamples < Int.MaxValue`? :)